src/librustc_mir/hair/cx/expr.rs

```diff
@@ -60,6 +60,15 @@ impl<'tcx> Mirror<'tcx> for &'tcx hir::Expr {
                     kind: ExprKind::UnsafeFnPointer { source: expr.to_ref() },
                 };
             }
+            Some(&ty::adjustment::AdjustEmptyToAny(..)) => {
+                let adjusted_ty = cx.tcx.expr_ty_adjusted(self);
```

Comment:
Ah, this and that other bug were because I took the type out of `AdjustEmptyToAny` and put it back in later.